akka-stream/src/main/scala/akka/stream/javadsl/Sink.scala

```diff
@@ -31,6 +32,31 @@ object Sink {
     new Sink(scaladsl.Sink.fold[U, In](zero)(f.apply).toCompletionStage())
 
   /**
+   * A `Sink` that will use Java 8 `Collector` to reduce and optionally transform
+   * accumulated result into a final representation.
+   * As collector is mutable you can use it only with one stream.
+   * Reduction operations performs sequentially.
+   */
+  def javaCollector[T, A, R](collector: java.util.stream.Collector[T, A, R]): Sink[T, CompletionStage[R]] =
+    new Sink(scaladsl.Sink.javaCollector[T, A, R](collector).toCompletionStage())
```

Comment:
um, we discussed this at an Akka Team meeting, but there were a lot of different opinions (naming). I thought we had decided on `StreamConverters`, see this comment: https://github.com/akka/akka/pull/19456#issuecomment-171678612, but it seems like there are still different opinions. We should not rush this one to tomorrow's release, so let's continue discussion next week.